### PR TITLE
Closing resources properly and another minor code warnings removal

### PR DESCRIPTION
**Pull request by garcia-jj**
Using try-close-resources to reduce some Eclipse warnings and other minor code adjustments.

### CODE REVIEW

File: vraptor-core/src/main/java/br/com/caelum/vraptor/cache/CacheStoreFactory.java [hunk 1]
```diff
@@ -32,13 +32,13 @@
 	@Produces
 	@Default
 	public <K, V> DefaultCacheStore<K, V> buildDefaultCache() {
-		return new DefaultCacheStore<K, V>();
+		return new DefaultCacheStore<>();
```
Comment (Turini) [inline vraptor-core/src/main/java/br/com/caelum/vraptor/cache/CacheStoreFactory.java]: is it valid for java 7? i cant test it now, but i'm almost sure that it isn't

File: vraptor-core/src/main/java/br/com/caelum/vraptor/cache/CacheStoreFactory.java [hunk 2]
```diff
@@ -32,13 +32,13 @@
...
 	}
 
 	@Produces
 	@LRU
 	public <K, V> LRUCacheStore<K, V> buildLRUCache(InjectionPoint ip) {
 		int capacity = ip.getAnnotated().getAnnotation(LRU.class).capacity();
-		return new LRUCacheStore<K, V>(capacity);
+		return new LRUCacheStore<>(capacity);
```
Comment (Turini) [inline vraptor-core/src/main/java/br/com/caelum/vraptor/cache/CacheStoreFactory.java]: same here

File: vraptor-core/src/main/java/br/com/caelum/vraptor/cache/CacheStoreFactory.java (hunk 1 above)
Comment (garcia-jj) [inline vraptor-core/src/main/java/br/com/caelum/vraptor/cache/CacheStoreFactory.java]: Diamonds are introduced on Java 7. But I'll check using a Java 7 compiler.

Comment (garcia-jj) [inline vraptor-core/src/main/java/br/com/caelum/vraptor/cache/CacheStoreFactory.java]: Yep, works on Java 7. I use this change to test:

```
diff --git a/vraptor-core/pom.xml b/vraptor-core/pom.xml
index 7ab4f4d..0adcd11 100644
--- a/vraptor-core/pom.xml
+++ b/vraptor-core/pom.xml
@@ -244,6 +244,7 @@
                                <artifactId>maven-compiler-plugin</artifactId>
                                <version>3.1</version>
                                <configuration>
+                                       <executable>/opt/jdk1.7.0_45/bin/javac</executable>
                                        <source>1.7</source>
                                        <target>1.7</target>
                                </configuration>
```

**Pull request closed without merging**